### ISSUE TITLE
MP 5.2 causes SharedLibrary opt-out

### ISSUE THREAD

**Issue by MattWhilden** [dotnet/corefx#9711]
The new metapackage seems incompatible with our ShareLibrary checks due to pulling in a newer System.Private.URI. 

Repro:
*new UWP project
*update the MP to 5.2
*build

Output:
1>  This project has been automatically opted out of the 'Shared Framework' app-size optimization, because it directly or indirectly references an assembly which is incompatible with the optimization. 'C:\Users\mwhilden\.nuget\packages\runtime.win7.System.Private.Uri\4.0.1\runtimes\aot\lib\netcore50\System.Private.Uri.dll', found version = '4.0.1', needed version = '4.0.0.0'.   (TaskId:252)

This, coupled with a bug in the .NET Native targets authoring will cause UWP submission to store to fail. I also think it's a bit odd that this opt-out is issued at Information level and not Warning+. Will probably get that changed on our side.

@weshaggard @ericstj

**Comment by weshaggard** [comment]
That is very unfortunate and we will look at fixing this. If anyone hits this a temporary workaround would be to add the following to their dependencies in project.json:

```
    "runtime.win7.System.Private.Uri": {
      "version": "4.0.1",
      "exclude": "runtime"
    },
    "System.Private.Uri": "4.0.0"
```

**Comment by MattWhilden** [comment]
The workaround above has the side effect of hundreds of build warnings which may be unsavory to some. Here's another method to avoid the problem. Add this to your project to disable SharedLibrary entirely:
<UseDotNetNativeSharedAssemblyFrameworkPackage>false</UseDotNetNativeSharedAssemblyFrameworkPackage>

**Comment by ericstj** [comment]
@MattWhilden that workaround turns off shared library, which is not what we want.  Wes's workaround will still permit use of shared library.

**Comment by MattWhilden** [comment]
That's a fair point but I wasn't convinced that the addition of 324 warnings to the empty UWP was worth leaving SharedLibrary enabled. I suppose folks could choose what works best for them. It looks like it's being resolved shortly so probably not worth hashing out which is "better". :-)

**Comment by ericstj** [comment]
Reactivating for fix in release/1.0.0

**Comment by ericstj** [comment]
This update has been pushed to nuget.org. https://www.nuget.org/packages/Microsoft.NETCore.UniversalWindowsPlatform/5.2.1

**Comment by ericstj** [comment]
Looks like we need another fix here.  https://www.myget.org/feed/ericstj-staging/package/nuget/Microsoft.NETCore.UniversalWindowsPlatform/5.2.2 has that fix.  Working on release.

**Comment by ericstj** [comment]
This fix was released to nuget.org.

**Comment by andrey-denisenko** [comment]
Hello There!
I faced with this issue when I tried to use **System.IO.Compression.ZipArchive** class.

- Visual Studio 2015
- target Windows 10 Anniversary Edition (10.0; Build 14393)

And as here was described, this problem appears only with ARM version of the app, and only when I trying to use shared libraries ("Compile with .NET Native toolchain" is UNchecked).

----- PROJECT.JSON -----------
{
  "dependencies": {
    "Microsoft.EntityFrameworkCore.Sqlite": "1.1.1",
    "Microsoft.EntityFrameworkCore.Tools": "1.1.0",
    "Microsoft.NETCore.UniversalWindowsPlatform": "**5.2.3**",
    "Microsoft.Xaml.Behaviors.Uwp.Managed": "2.0.0",
    "MvvmLight": "5.3.0"
  },
  "frameworks": {
    "uap10.0": {}
  },
  "runtimes": {
    "win10-arm": {},
    "win10-arm-aot": {},
    "win10-x86": {},
    "win10-x86-aot": {},
    "win10-x64": {},
    "win10-x64-aot": {}
  }
}
------------------------------

![2017-04-17 1](https://cloud.githubusercontent.com/assets/26914037/25111946/157d8fc4-23bc-11e7-9caa-c997aa4de9b2.png)
![2017-04-17 2](https://cloud.githubusercontent.com/assets/26914037/25111947/157dda88-23bc-11e7-8ac3-a12da554b203.png)

Is there any thoughts how to fix this without compiling with native tool chain? Thanks!
